strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout/strongbox-npm-metadata-jackson/src/main/java/org/carlspring/strongbox/npm/metadata/jackson/EnginesDeserializer.java

```diff
@@ -0,0 +1,42 @@
+package org.carlspring.strongbox.npm.metadata.jackson;
+
+import java.io.IOException;
+
+import org.carlspring.strongbox.npm.metadata.Engines;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class EnginesDeserializer extends JsonDeserializer<Engines>
+{
+
+    @Override
+    public Engines deserialize(JsonParser jp,
+                               DeserializationContext c)
+        throws IOException,
+        JsonProcessingException
+    {
+
+        ObjectCodec codec = jp.getCodec();
+        JsonNode node = codec.readTree(jp);
+
+        String value = node.asText();
+
+        if (value.startsWith("{"))
+        {
+            return codec.treeToValue(node, Engines.class);
+        }
+
+        Engines engines = new Engines();
+        // TODO: we should replace this {packageName} placeholer with package
+        // name.
```

Comment:
@sbespalov : Is this something we need to add a follow-up task for, or is this no longer relevant (in which case we can drop the comment)?

Reply:
@sbespalov ping? :)